src/ess/livedata/dashboard/roi_publisher.py

```diff
@@ -4,17 +4,25 @@
 
 import logging
 
-from ..config.models import RectangleROI
-from ..config.roi_names import get_roi_mapper
+from ..config.models import PolygonROI, RectangleROI
 from ..config.workflow_spec import JobId
 from ..core.message import Message, MessageSink, StreamId, StreamKind
 
+ROIs = dict[int, RectangleROI] | dict[int, PolygonROI]
+ROIClass = type[RectangleROI] | type[PolygonROI]
+
+# Map ROI class to stream key suffix
+_READBACK_KEYS: dict[ROIClass, str] = {
+    RectangleROI: "roi_rectangle",
+    PolygonROI: "roi_polygon",
+}
```

Comment:
These should be defined in terms of `ROIGeometry.readback_key`. In general, I think there should be some association between the ROI models and `ROIGeometry` so we don't rely on ad hoc associations scattered throughout the code base.